actioncable/app/assets/javascripts/action_cable/connection.coffee

```diff
@@ -74,10 +82,12 @@ class ActionCable.Connection
 
   events:
     message: (event) ->
+      return if @disconnected
```

Comment:
Yes the server fires out the welcome message pretty fast after connecting. The server doesn't decide to disconnect on a protocol mismatch, the client does.